### ISSUE TITLE
openssl doesn't build on aix in OpenJ9 builds

### ISSUE THREAD

**Issue by pshipton** [ibmruntimes/openj9-openjdk-jdk8#123]
Re #112

On aix using
get_source.sh --openssl-version=1.1.1
configure with --with-openssl=fetched --enable-openssl-bundling

```
15:36:02 ar: creating an archive file apps/libapps_a.a
15:36:19      147  1500-010: (W) WARNING in async_start_func: Infinite loop.  Program may not stop.
15:37:03 "crypto/ec/ecp_nistz256.c", line 227.31: 1506-1298 (W) The subscript 4 is out of range. The valid range is 0 to 3.
15:37:03 "crypto/ec/ecp_nistz256.c", line 228.31: 1506-1298 (W) The subscript 5 is out of range. The valid range is 0 to 3.
15:37:03 "crypto/ec/ecp_nistz256.c", line 229.31: 1506-1298 (W) The subscript 6 is out of range. The valid range is 0 to 3.
15:37:03 "crypto/ec/ecp_nistz256.c", line 1251.37: 1506-1298 (W) The subscript 4 is out of range. The valid range is 0 to 3.
15:37:03 "crypto/ec/ecp_nistz256.c", line 1251.48: 1506-1298 (W) The subscript 5 is out of range. The valid range is 0 to 3.
15:37:03 "crypto/ec/ecp_nistz256.c", line 1251.59: 1506-1298 (W) The subscript 6 is out of range. The valid range is 0 to 3.
15:37:03 "crypto/ec/ecp_nistz256.c", line 1251.70: 1506-1298 (W) The subscript 7 is out of range. The valid range is 0 to 3.
15:37:03 "crypto/ec/ecp_nistz256.c", line 1252.37: 1506-1298 (W) The subscript 4 is out of range. The valid range is 0 to 3.
15:37:03 "crypto/ec/ecp_nistz256.c", line 1252.48: 1506-1298 (W) The subscript 5 is out of range. The valid range is 0 to 3.
15:37:03 "crypto/ec/ecp_nistz256.c", line 1252.59: 1506-1298 (W) The subscript 6 is out of range. The valid range is 0 to 3.
15:37:03 "crypto/ec/ecp_nistz256.c", line 1252.70: 1506-1298 (W) The subscript 7 is out of range. The valid range is 0 to 3.
15:37:03 "crypto/ec/ecp_nistz256.c", line 1262.36: 1506-1298 (W) The subscript 4 is out of range. The valid range is 0 to 3.
15:37:03 "crypto/ec/ecp_nistz256.c", line 1262.27: 1506-1298 (W) The subscript 4 is out of range. The valid range is 0 to 3.
15:37:03 "crypto/ec/ecp_nistz256.c", line 1263.36: 1506-1298 (W) The subscript 5 is out of range. The valid range is 0 to 3.
15:37:03 "crypto/ec/ecp_nistz256.c", line 1263.27: 1506-1298 (W) The subscript 5 is out of range. The valid range is 0 to 3.
15:37:03 "crypto/ec/ecp_nistz256.c", line 1264.36: 1506-1298 (W) The subscript 6 is out of range. The valid range is 0 to 3.
15:37:03 "crypto/ec/ecp_nistz256.c", line 1264.27: 1506-1298 (W) The subscript 6 is out of range. The valid range is 0 to 3.
15:37:03 "crypto/ec/ecp_nistz256.c", line 1265.36: 1506-1298 (W) The subscript 7 is out of range. The valid range is 0 to 3.
15:37:03 "crypto/ec/ecp_nistz256.c", line 1265.27: 1506-1298 (W) The subscript 7 is out of range. The valid range is 0 to 3.
15:37:13     1500-030: (I) INFORMATION: openssl_add_all_ciphers_int: Additional optimization may be attained by recompiling and specifying MAXMEM option with a value greater than 16384.
15:38:36 ar: creating an archive file libcrypto_a.a
15:38:57 ar: creating an archive file libssl_a.a
15:38:59 ar: creating an archive file test/libtestutil_a.a
15:39:03 ar: creating an archive file libcrypto.a
15:39:06 ar: creating an archive file libssl.a

...

16:08:21 Compiling 226 files for BUILD_SECURITY
16:08:28 Note: Some input files use or override a deprecated API.
16:08:28 Note: Recompile with -Xlint:deprecation for details.
16:08:33 gmake[2]: *** No rule to make target '/home/u0020236/workspace/PullRequest-Compile-JDK8-aix_ppc-64_cmprssptrs-OpenJ9/openssl/libcrypto.so.1.1', needed by '/home/u0020236/workspace/PullRequest-Compile-JDK8-aix_ppc-64_cmprssptrs-OpenJ9/build/aix-ppc64-normal-server-release/jdk/lib/ppc64/libcrypto.so.1.1'.  Stop.
16:08:33 gmake[2]: *** Waiting for unfinished jobs....
16:08:33 BuildJdk.gmk:74: recipe for target 'libs-only' failed
16:08:33 gmake[1]: *** [libs-only] Error 2
16:08:33 /home/u0020236/workspace/PullRequest-Compile-JDK8-aix_ppc-64_cmprssptrs-OpenJ9//make/Main.gmk:119: recipe for target 'jdk-only' failed
16:08:33 make: *** [jdk-only] Error 2
```
https://ci.eclipse.org/openj9/job/PullRequest-Compile-JDK8-aix_ppc-64_cmprssptrs-OpenJ9/36/console

**Comment by pshipton** [comment]
@mbvreddy

**Comment by mbvreddy** [comment]
OpenSSL 1.1.1 crypto library name has changed. Due to which the issue is seen when --enable-openssl-bundling is used.

I will push the fix.

**Comment by mbvreddy** [comment]
Directory structure in OpenSSL 1.1.0i
<img width="565" alt="screen shot 2018-10-08 at 12 19 00 pm" src="https://user-images.githubusercontent.com/13833097/46595734-7e548000-caf7-11e8-9f5b-ecedd4c9fad1.png">

Directory structure in OpenSSL 1.1.1
<img width="473" alt="screen shot 2018-10-08 at 12 20 24 pm" src="https://user-images.githubusercontent.com/13833097/46595745-8dd3c900-caf7-11e8-9ca0-7df9c6d0852f.png">

Linker options -lcrypto indeed link it against libcrypto.a. And, it is expected that libcrypto.a that has libcrypto64.so.1.1 is present in LIBPATH. Its bit tricky to handle this situation.

**Comment by mbvreddy** [comment]
Created PR https://github.com/ibmruntimes/openj9-openjdk-jdk8/pull/124

**Comment by pshipton** [comment]
Fixed by #124
